python/ray/air/result.py

```diff
@@ -166,42 +170,46 @@ def from_path(cls, path: str) -> "Result":
         latest_metrics = metrics_df.iloc[-1].to_dict() if not metrics_df.empty else {}
 
         # Restore all checkpoints from the checkpoint folders
-        ckpt_dirs = [
-            join(local_path, entry)
-            for entry in file_list
-            if entry.startswith("checkpoint_")
-        ]
-
-        ckpt_metadata_dicts = [
-            ray.cloudpickle.load(
-                open(join(ckpt_dir, CHECKPOINT_TUNE_METADATA_FILE), "rb")
-            )
-            for ckpt_dir in ckpt_dirs
-        ]
-
-        if ckpt_dirs:
-            checkpoints = [
-                Checkpoint.from_directory(ckpt_dir) for ckpt_dir in ckpt_dirs
-            ]
-
-            checkpoint_ids = [
-                metadata["iteration"] - 1 for metadata in ckpt_metadata_dicts
-            ]
-
-            checkpoint_metrics = [
-                metadata["last_result"] for metadata in ckpt_metadata_dicts
-            ]
-
-            # TODO(air-team): make metrics a property of Checkpoint
-            best_checkpoints = list(zip(checkpoints, checkpoint_metrics))
-            latest_checkpoint_index = checkpoint_ids.index(max(checkpoint_ids))
-            latest_checkpoint = checkpoints[latest_checkpoint_index]
+        checkpoint_dirs = sorted(local_path.glob("checkpoint_*"))
+
+        if checkpoint_dirs:
+            if _use_storage_context():
+                checkpoints = [
+                    NewCheckpoint.from_directory(checkpoint_dir)
+                    for checkpoint_dir in checkpoint_dirs
+                ]
+            else:
+                checkpoints = [
+                    Checkpoint.from_directory(checkpoint_dir)
+                    for checkpoint_dir in checkpoint_dirs
+                ]
+
+            metrics = []
+            for checkpoint_dir in checkpoint_dirs:
+                metrics_corresponding_to_checkpoint = metrics_df[
+                    metrics_df[CHECKPOINT_DIR_NAME] == checkpoint_dir.name
+                ]
```

Comment:
Oh, the `metrics_df` got loaded from the json, and I'm just reading the checkpoint dir name column from it to match w/ the checkpoint dir name.

Reply:
@ericl

Reply:
Oh, got it!